src/HIE/Bios/Cradle.hs

```diff
@@ -466,13 +470,53 @@ cabalProcess workDir command args = do
   let newEnvironment = extraEnvironmentArgs ++ environment
   wrapper_fp <- withCabalWrapperTool ("ghc", []) workDir
   buildDir <- cabalBuildDir workDir
-  let cabalArgs = ["--builddir=" <> buildDir, command, "--with-compiler", wrapper_fp] ++ args
-  let cabalProc = proc "cabal" cabalArgs
+  let baseCabalArgs = ["--builddir=" <> buildDir, command, "--with-compiler", wrapper_fp]
+  extraCabalArgs <- case ghcDirLoadResult of
+        CradleSuccess (ghcBin, libdir) -> do
+          ghcPkgPath <- withGhcPkgTool ghcBin libdir
+          pure ["--with-hc-pkg", ghcPkgPath]
+        _ -> pure []
+  let cabalProc = proc "cabal" (baseCabalArgs ++ extraCabalArgs ++ args)
   pure $ CradleSuccess (cabalProc
       { env = Just newEnvironment
       , cwd = Just workDir
       })
 
+withGhcPkgTool :: FilePath -> FilePath -> IO FilePath
+withGhcPkgTool ghcPathAbs libdir = do
+  let ghcName = takeFileName ghcPathAbs
+  -- TODO: check for existence
+  let ghcPkgPath = guessGhcPkgFromGhc ghcName
+  if isWindows
+    then pure ghcPkgPath
+    else withWrapperTool ghcPkgPath
+  where
+    ghcDir = takeDirectory ghcPathAbs
+
+    guessGhcPkgFromGhc ghcName =
+      case parseCompilerId (T.pack ghcName) of
+        Nothing -> ghcDir </> "ghc-pkg"
+        Just version -> ghcDir </> "ghc-pkg-" <> showVersion version
+
+    withWrapperTool ghcPkg = do
+      let globalPackageDb = libdir </> "package.conf.d"
+      let contents = unlines
+            [ "#!/bin/sh"
+            , unwords ["command", ghcPkg, "--global-package-db", globalPackageDb, "${1+\"$@\"}"]
+            ]
```

Comment:
Should this be documented, or extraced just like the cabal wrapper?

Reply:
Def. docs